src/Psalm/Internal/Analyzer/ProjectAnalyzer.php

```diff
@@ -716,9 +716,7 @@ public function interpretRefactors() : void
 
                 $source_class_storage = $this->codebase->classlike_storage_provider->get($source_parts[0]);
 
-                $destination_parts = explode('\\', $destination);
-
-                array_pop($destination_parts);
+                $destination_parts = explode('\\', $destination, -1);
```

Comment:
explode has a param which can be negative. If positive, it can reduce the resulting array, if negative, it will remove last elements. This allow dropping the array_pop